neware_parser/DataEngine.py

```diff
@@ -8,6 +8,38 @@
 
 class DataEngine:
 
+    @staticmethod
+    def compute_shift(
```

Comment:
The function that computes shift should be separate from the function that pickles the result.

In fact, you should compute everything that you need to compute for a given barcode by having something like compute_all_scalars() which calls compute_shift, compute_r, compute ... and build a dictionary or something like that with all those results 

The thing that pickles should be very simple and at the outer scope where you loop over barcodes. It should be responsible for sending the right data to the right file.